source/ec_override.c

```diff
@@ -208,7 +184,9 @@ int EC_KEY_up_ref(EC_KEY *key) {
  * zero then frees the memory associated with it. If key is NULL nothing is done.
  */
 void EC_KEY_free(EC_KEY *key) {
-    if (key) {
+    if (key != NULL &&
+        /* We must include this extra guard to avoid spurious arithmetic underflows. */
+        key->references >= 0) {
```

Comment:
Should this be `> 0`? Otherwise the `-= 1` below might still underflow, right?

And we're only checking `==0` afterwards, so either that could be made `<= 0` or this should be `> 0`, right?

Reply:
You're correct, this should be `>0` to properly avoid overflows.